### ISSUE TITLE
TypeScript type definitions

### ISSUE THREAD

**Issue by garyking** [bttmly/nba#83]
It would be great to have TypeScript type definitions for this library. Thanks!

**Comment by bttmly** [comment]
doing the argument signature would be doable but getting types on the API response shapes would be a lot of work without some kind of tool to automate it. I don't have a lot of time at the moment, but might take a stab at this later, or PRs are welcome!

**Comment by garyking** [comment]
I've created types just for the methods I'm using, for myself. I use http://json2ts.com/ to create the API response types.

**Comment by bttmly** [comment]
Yeah I've noticed that they seem to have rather lax standards for API development and consistency.

**Comment by bttmly** [comment]
@garyking do you have any experience writing `d.ts` files for javascript libraries? I started a [branch to add strong types](https://github.com/bttmly/nba/pull/91) but I can't seem to figure out how to export what I want. 

If you check out that branch, you can generate the underlying types for each data endpoint with this:
```
WRITE_RESPONSES=1 yarn run mocha test/integration/data.js; node ./scripts/makeTypes.js
```
